src/test/java/org/example/controller/ApplicationControllerTests.java

```diff
@@ -0,0 +1,84 @@
+package org.example.controller;
+
+import org.example.controllers.ApplicationController;
+import org.example.exceptions.DatabaseConnectionException;
+import org.example.exceptions.DoesNotExistException;
+import org.example.exceptions.Entity;
+import org.example.exceptions.InvalidException;
+import org.example.models.ApplicationRequest;
+import org.example.services.ApplicationService;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import javax.ws.rs.core.Response;
+import java.sql.SQLException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ApplicationControllerTests {
+
+    ApplicationService applicationService =
+            Mockito.mock(ApplicationService.class);
+
+    ApplicationController applicationController =
+            new ApplicationController(applicationService);
+
+    private static final ApplicationRequest applicationRequest =
+            new ApplicationRequest(
+                    "adam@random.com",
+                    2,
+                    "https://www.youtube.com/watch?v=dQw4w9WgXcQ"
+            );
+
+    @Test
+    void apply_shouldReturn201_whenSuccessful() {
+
+
+        Response response = applicationController.apply(applicationRequest);
```

Comment:
how is this working without mocking out the service?

Reply:
In all honesty, i think it skipped it because i create a mocked variable at the top of the file. It should be fixed in the latest push